app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/configuration/ConnectingDevicesFragment.kt

```diff
@@ -85,15 +94,11 @@ class ConnectingDevicesFragment : BaseDaggerFragment(), NsdServiceManager.OnServ
         showSnackbarMessage(R.string.discovery_start_failed)
     }
 
-    private fun handleAppState(state: AppState?) {
-        when (state) {
-            AppState.CLIENT -> {
-                findNavController().navigate(R.id.configurationToClientHome)
-                requireActivity().finish()
-            }
-            else -> {
-                findNavController().navigate(R.id.configurationToAllDone)
-            }
+    private fun onConnectionCompleted(connectionCompleted: Boolean) {
+        if (connectionCompleted) {
+            findNavController().navigate(R.id.connectingDevicesToAllDone)
+        } else {
+            findNavController().navigate(R.id.connectionFailed)
         }
```

Comment:
I think that this would look better if you return th navigation id from if statement.
```suggestion
        findNavController().navigate(if (connectionCompleted) {
            R.id.connectingDevicesToAllDone
        } else {
            R.id.connectionFailed
        })
```